### ISSUE TITLE
PopupView positions incorrectly with multiple displays 

### ISSUE THREAD

**Issue by frmdstryr** [nucleic/enaml#499]
The `ensure_on_screen` function in https://github.com/nucleic/enaml/blob/main/enaml/qt/q_popup_view.py#L866 seems to mess up the popup location when multiple screens are present (at least on linux with Qt6, didn't test anything else). 

I have 3 screens and it places the popup on the first leftmost screen when the window is on the rightmost screen.  If that function is commented out then it works as I would expect.

**Comment by frmdstryr** [comment]
Perhaps it can be disabled on linux? I added a few print statements

```python
target_pos=PyQt6.QtCore.QPoint(4800, 750)
anchor_pos=PyQt6.QtCore.QPoint(100, 0)
trial_geo=PyQt6.QtCore.QRect(4700, 720, 200, 69) # This is the correct position on my screen
desk_geo=PyQt6.QtCore.QRect(0, 0, 1920, 1080)
geo=PyQt6.QtCore.QRect(1720, 720, 200, 69)
```

I'm curious if desk_geo returns all monitors as one big screen on windows? At least on my setup it appears to be clamping it to within a single screen size (my setup is 3 FHD screens in a horizontal line).

There is a note in QScreen https://doc.qt.io/qt-5/qscreen.html#availableGeometry-prop
> Note, on X11 this will return the true available [geometry](https://doc.qt.io/qt-5/qscreen.html#geometry-prop) only on systems with one monitor and if window manager has set _NET_WORKAREA atom. In all other cases this is equal to geometry(). This is a limitation in X11 window manager specification.

**Comment by MatthieuDartiailh** [comment]
Closed by #500
